### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by davidcollom**

### CODE REVIEW

**Review by hawksight — APPROVED** [review]
Based on Dockerfile changes and the go.mod changes (top of file), this all seems sensible and more recent 👍 

Assuming most of the rest is the work of Gophers :)

**Comment by hawksight** [comment]
@davidcollom 👍  but conflicts ^

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
+FROM --platform=$BUILDPLATFORM golang:1.22-alpine as builder
```
Comment (davidcollom) [inline Dockerfile:1]:
```suggestion
FROM --platform=$BUILDPLATFORM golang:1.22-alpine AS builder
```

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
+FROM --platform=$BUILDPLATFORM golang:1.22-alpine AS builder
 
 RUN apk --no-cache add make
```
Comment (davidcollom) [inline Dockerfile:4]:
```suggestion
FROM --platform=$BUILDPLATFORM golang:1.22-alpine AS builder

```

**Pull request merged**